pkg/didcomm/protocol/didexchange/keys.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package didexchange
+
+import (
+	"encoding/json"
+	"fmt"
+
+	gojose "github.com/square/go-jose/v3"
+
+	"github.com/hyperledger/aries-framework-go/pkg/crypto"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+)
+
+func createNewKeyAndVM(didDoc *did.Doc, keyType, keyAgreementType kms.KeyType, keyManager kms.KeyManager) error {
+	vm, err := createSigningVM(keyManager, getVerMethodType(keyType), keyType)
+	if err != nil {
+		return err
+	}
+
+	kaVM, err := createEncryptionVM(keyManager, getVerMethodType(keyAgreementType), keyAgreementType)
+	if err != nil {
+		return err
+	}
+
+	didDoc.VerificationMethod = append(didDoc.VerificationMethod, *vm)
+
+	didDoc.Authentication = append(didDoc.Authentication, *did.NewReferencedVerification(vm, did.Authentication))
+	didDoc.KeyAgreement = append(didDoc.KeyAgreement, *did.NewReferencedVerification(kaVM, did.KeyAgreement))
+
+	return nil
+}
+
+func createSigningVM(km kms.KeyManager, vmType string, keyType kms.KeyType) (*did.VerificationMethod, error) {
+	kid, pubKeyBytes, err := km.CreateAndExportPubKeyBytes(keyType)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Might be a good idea to wrap these errors

Reply:
done